File: PHP/Queue.php
<?php
class Node {
    function __contruct($data, $node){
        $this->data = $data;
        $this->link = $node;
    }

    function setLink($node){
        $this->link = $node;
    }

    function getLink(){
        return $this->link;
    }
}

class Queue {
    function __contruct(){
        $this->front = null;
        $this->back = null;
    }

    function enqueue($data){
        $temp = new Node($data, null);
        if ($this->front === null){
            $this->back = $temp;
            $this->front = $this->back;
        } else {
            $temp->link = $this->back;
            $this->back = $temp;
        }
    }

    function peek(){
        if ($this->front === null){
            echo "Queue Empty!";
        }
        return $this->front->data;
    }

    function dequeue(){
        if ($this->front === null){
            echo "Queue Empty!";
        }
        $data = $this->front->data;
        $this->front = $this->front->link;
        return $data;
    }

    function isEmpty(){
        if ($this->front === null){
            return true;
        } else {
            return false;
        }
    }

    function size(){
        $temp = $this->front;
        $counter = 0;
        while($temp !== null){
            $counter++;
            $temp = $temp->link;
        }
        return $counter;
    }

    function printList(){
        $temp = $this->front;
        while($temp !== null){
            echo $temp->data;
            $temp = $temp->link;
        }
    }
}

$queue = new Queue();

echo $queue->isEmpty();

$queue->enqueue(1);
$queue->enqueue(3);
$queue->enqueue(5);
$queue->enqueue(7);
$queue->enqueue(9);

echo $queue->peek();

$queue->printList();
echo $queue->dequeue();
echo $queue->isEmpty();
$queue->printList();

?>

File: Java/Queue.java
public class Queue {

	Node front = null;
	Node rear = null;

	public class Node {
		int data
		Node link;

		Node(val, node){
			data = val;
			link = node;
		}

		public void setLink(nodeLink){
			link = nodeLink;
		}

		public Node getLink(){
			return link;
		}
	}

	public void enqueue(int data){
		Node temp = new Node(data, null);
		if (front == null){
			front = temp;
			rear = temp;
		}
		else {
			rear.setLink(temp);
			rear = temp;
		}
	}

	public int peek(){
		if (front != null){
			return front.data;
		}
	}

	public void dequeue(){
		return front = front.link;
	}

	public int size(){
		Node temp = front;
		int counter = 0;
		while(temp != null){
			temp = temp.link;
			counter++;
		}
		return counter;
	}

	public void printList(){
		Node temp = front;
		while(temp.link != null){
			system.out.println(temp.data);
			temp = temp.link;
		}
	}

	public static void main(String [] args){
		System.out.println("Hello There!");

		Queue queue = new Queue();
		System.out.println(queue.isEmpty());

		queue.push(1);
		queue.push(3);
		queue.push(5);
		queue.push(7);
		queue.push(9);

		queue.printList();
		System.out.println(queue.isEmpty());	
		System.out.println(queue.peek());
		queue.pop();

		System.out.println("Hello World!");
	}
}
